JhipsterXamarin/JhipsterXamarin/JhipsterXamarin/ViewModels/RegisterViewModel.cs

```diff
@@ -0,0 +1,212 @@
+            set
+            {
+                _error = value;
+                RaisePropertyChanged(() => Error);
+            }
+        }
+
+        public string ConfirmPassword
+        {
+            get => _confirmPassword;
+            set
+            {
+                _confirmPassword = value;
+                RaisePropertyChanged(() => ConfirmPassword);
+                ReloadActive();
+            }
+        }
+
+        public bool ErrorMail
+        {
+            get => _errorMail;
+            set
+            {
+                _errorMail = value;
+                RaisePropertyChanged(() => ErrorMail);
+            }
+        }
+
+        public bool ErrorLogin
+        {
+            get => _errorLogin;
+            set
+            {
+                _errorLogin = value;
+                RaisePropertyChanged(() => ErrorLogin);
+            }
+        }
+
+        public bool Success
+        {
+            get => _success;
+            set
+            {
+                _success = value;
+                RaisePropertyChanged(() => Success);
+                RaisePropertyChanged(() => NotSuccess);
+            }
+        }
+
+        public bool NotSuccess
+        {
+            get => !_success;
+        }
+
+        public RegisterViewModel(IMvxNavigationService navigationService, IAuthenticationService authenticationService, IRegisterService registerService)
+        {
+            _navigationService = navigationService;
+            _authenticationService = authenticationService;
+            _registerService = registerService;
+
+            Error = false;
+
+            SignUp = new MvxCommand(async () =>
+            {
+                Active = false;
+                var resultError = await _registerService.Save(new UserSaveModel
+                {
+                    Password = Password,
+                    Login = Username,
+                    Email = Email,
+                    LangKey = "en"
+                });
+
+                Error = resultError == ErrorConst.ProblemBaseUrl;
+                ErrorLogin = resultError == ErrorConst.LoginAlreadyUsedType;
+                ErrorMail = resultError == ErrorConst.EmailAlreadyUsedType;
+
+                Success = (!new List<bool>() { Error, ErrorLogin, ErrorMail }.Contains(true));
+            });
+
+            GoBack = new MvxCommand(() =>
+            {
+                _navigationService.Navigate<LoginViewModel>();
+            });
+
+            ChangeStateCommand = new MvxCommand(() =>
+            {
+                RememberMe = !RememberMe;
+            });
+        }
+
+        public void ReloadActive()
+        {
+            if (string.IsNullOrEmpty(Password) || string.IsNullOrEmpty(Username))
+                Active = false;
+            else
+            {
+                var emailSplit = Email.Split('@');
+                Active = Password.Length > 3 && (ConfirmPassword == Password) && Email.Length > 4 && emailSplit.Length == 2 &&
```

Comment:
too much conditions

Reply:
Moved those conditions into a method in commit 6e4406d